tls/s2n_resume.c

```diff
@@ -284,11 +284,13 @@ int s2n_connection_get_session_ticket_lifetime_hint(struct s2n_connection *conn)
     return conn->ticket_lifetime_hint;
 }
 
-int s2n_connection_get_session_length(struct s2n_connection *conn)
+uint32_t s2n_connection_get_session_length(struct s2n_connection *conn)
 {
     /* Session resumption using session ticket "format (1) + session_ticket_len + session_ticket + session state" */
     if (conn->config->use_tickets && conn->client_ticket.size > 0) {
-        return S2N_STATE_FORMAT_LEN + S2N_SESSION_TICKET_SIZE_LEN + conn->client_ticket.size + S2N_STATE_SIZE_IN_BYTES;
+        uint32_t result;
+        GUARD(s2n_add_overflow(S2N_STATE_FORMAT_LEN + S2N_SESSION_TICKET_SIZE_LEN + S2N_STATE_SIZE_IN_BYTES, conn->client_ticket.size, &result));
```

Comment:
You should only be using `GUARD` inside a function that returns `int`, which is really the error signal in s2n. Otherwise it returns `-1` which is cast into the max value of `uint32_t` which can cause some subtle bugs. The more appropriate API signature would be:

```C
int s2n_connection_get_session_length(struct s2n_connection *conn, uint32_t *length);
```

Reply:
BTW, this is a good example of how #1782 would improve `GUARD` usability.

Reply:
Thanks for the explanation